server/world/explosion/source.go

```diff
@@ -0,0 +1,35 @@
+package explosion
+
+import "github.com/df-mc/dragonfly/server/world"
+
+// Source is the source of the explosion. This can range from creepers, TNT, and other sources.
+type Source interface {
+	// ExplodeInfo returns the explosion information for the source.
+	ExplodeInfo() *Explosion
+}
+
+// TNTSource is an explosion source that is caused by TNT.
+type TNTSource struct {
```

Comment:
I wouldn't implement these here but rather in the blocks or entities themselves.